src/lib/types.hpp

```diff
@@ -232,6 +232,8 @@ PredicateCondition inverse_predicate_condition(const PredicateCondition predicat
 //                      dropped. This behavior mirrors NOT EXISTS
 enum class JoinMode { Inner, Left, Right, FullOuter, Cross, Semi, AntiNullAsTrue, AntiNullAsFalse };
 
+enum class JoinInputSide { Left, Right };
```

Comment:
Sorry I didn't bring this up before: aren't both sides inputs? Make it a class member of the join and name it IndexedSide?